cmd/api/routers/routers_test.go

```diff
@@ -31,7 +32,39 @@ func TestGetAllResources(t *testing.T) {
 	if err := json.NewDecoder(res.Body).Decode(&resources); err != nil {
 		t.Fail()
 	}
-	assert.Equal(t, resources.Kind, "crontab")
-	assert.Equal(t, resources.APIVersion, "stable.example.com/v1")
+	assert.Equal(t, "crontabs", resources.Kind)
+	assert.Equal(t, "stable.example.com/v1", resources.APIVersion)
 	assert.Greater(t, len(resources.Items), 0)
 }
+
+func TestRouterInvalidMethods(t *testing.T) {
+	router := setupRouter()
+
+	tests := []struct {
+		name   string
+		method string
+	}{
+		{
+			name:   "Invalid POST",
+			method: "POST",
+		},
+		{
+			name:   "Invalid PUT",
+			method: "PUT",
+		},
+		{
+			name:   "Invalid DELETE",
+			method: "DELETE",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			res := httptest.NewRecorder()
+			req, _ := http.NewRequest(tc.method, "/apis/stable.example.com/v1/crontabs", nil)
+			router.ServeHTTP(res, req)
+
+			assert.Equal(t, http.StatusNotFound, res.Code, res.Result())
+		})
+	}
+}
```

Comment:
Does this work? You changed the name of `setupRouter()` function to `newServer()`.
This test tries out the router configuration, which is currently done on `main` package. With this refactor the logic is still on main, so I'd keep the test in `main_test.go` instead of `routers_test.go`.

Note that this isn't testing any of the logic defined on `routers/routers.go`

Reply:
Due to the inclusion of the AuthN middleware it was not working, so I misunderstood what it was trying to test. You are right in that is not testing anything on the router.go, its just testing that the GET method from the library works as expected.

With the inclusion of the AuthN middleware, all unautenticated requests are unauthorized before they are rejected for being an invalid verb. So the test is not valid anymore. I recreated as "TestAuthenticationMiddlewareIsConfigured".